cli/common/runtime.js

```diff
@@ -0,0 +1,9 @@
+// @ts-check
+import { createRequire } from "node:module";
+import * as path from "node:path";
+
+const require = createRequire(import.meta.url);
+
+const runtimePackageJson = require.resolve("@rescript/runtime/package.json");
```

Comment:
This looks like a mix of CommonJS and ESM. If the intent is to support older Node versions, please add a short comment explaining that and the design trade-offs. 

Also clarify why `oxc_resolver` Node module wasn't used here, so future readers know the exact rationale.